openhcl/hcl/src/ioctl.rs

```diff
@@ -1643,40 +1642,24 @@ pub enum NoRunner {
 }
 
 /// An isolation-type-specific backing for a processor runner.
-pub trait Backing<'a>: BackingPrivate<'a> {}
```

Comment:
`BackingPrivate` serves two purposes: it acts as a way to seal the trait so that it can't be implemented externally, and it hides the methods, so they can't be invoked externally.

With these changes, in practice, you still can't invoke methods externally. But I still prefer having the extra separation, so that you don't have to add `#[doc(hidden)]` to every method of `Backing` (which you have not done in this PR).

I like the other changes, to remove some of the pubs.